src/pages/Dashboard/Payment/index.js

```diff
@@ -1,60 +1,87 @@
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { toast } from 'react-toastify';
 import styled from 'styled-components';
 import BoxContainer from '../../../components/Dashboard/Containers/BoxContainer';
+import FlatButton from '../../../components/Dashboard/Containers/FlatButton';
+import DisplaySection from '../../../components/Dashboard/Sections/DisplaySection';
+import useTicket from '../../../hooks/api/useTicket';

```

Comment:
Carlos, seria interessante explicar o que os hooks fazem ajudaria entender;

Reply:
Claro, eu segui a mesma estrutura que o projeto já tinha. Pelo que entendi, os hooks são os encarregados para executar as requests descritas na camada de service, assim você não precisaria criar um useEffect especifico para poder realizar o request, assim você retira essa tarefa do componente.
Dentro dos hooks tem uma função chamada de useAsync, a qual recebe dois parametros (handler e immediate):
handler: é a função que ele deve realizar, no caso a requisição;
immediate: por default é true, o que faz que onde o "hook" estiver sendo utilizado, ao carregar o componente ele logo enseguida realizar a request para o endpoint, se for **false** ele não executa a request (Exemplo de uso: botões, formularios, etc...)